### PR TITLE
[DOC] Improve the documentation to better match CRAN requirements

### PR DESCRIPTION
**Pull request by csouchet**
Improve API documentation
Add executable examples

covers #10

### CODE REVIEW

File: R/bpmnVisualization.R [hunk 1]
```diff
@@ -12,16 +12,33 @@
...
+#'      
+#' @examples 
+#' # Load the BPMN file
+#' bpmn_file <- system.file("examples/Order_Management.bpmn", package = "bpmnVisualization")
+#' 
+#' # Display the BPMN diagram
+#' display(bpmn_file)
+#' 
+#' # Display the BPMN diagram with overlays
+#' overlays <- list(create_overlay("start_event_1_1", "42"), create_overlay("sequence_flow_1_1", "42"), create_overlay("task_1_1", "9"))
```
Comment (tbouffard) [inline R/bpmnVisualization.R:38]: ⚠️ This generated a NOTE in R-check
```
 Rd file 'display.Rd':
  \examples lines wider than 100 characters:
     overlays <- list(create_overlay("start_event_1_1", "42"), create_overlay("sequence_flow_1_1", "42"), create_overlay("task_1_1", "9"))
```

File: R/bpmnVisualization.R [hunk 2]
```diff
@@ -49,21 +73,18 @@ display <- function(
   )
 }
 
-#' Shiny bindings for bpmnVisualization
-#'
-#' Output and render functions for using bpmnVisualization within Shiny
-#' applications and interactive Rmd documents.
+#' @title Shiny output binding for 'bpmnVisualization' HTML widget
```
Comment (csouchet) [inline R/bpmnVisualization.R:76]:
```suggestion
#' @title Shiny output binding for \code{bpmn-visualization} HTML widget
```

File: R/bpmnVisualization.R [hunk 3]
```diff
@@ -12,16 +12,40 @@
...
-#' @param height The height used to display the widget
-#' @param elementId The id of the HTML element to enclose the widget
+#' @param width Fixed width for widget (in css units). The default is \code{NULL}, which results in intelligent automatic sizing based on the widget's container.
+#' @param height Fixed height for widget (in css units). The default is \code{NULL}, which results in intelligent automatic sizing based on the widget's container.
+#' @param elementId The id of the HTML element to enclose the widget.
+#'      Use an explicit element ID for the widget (rather than an automatically
+#'      generated one). Useful if you have other JavaScript that needs to explicitly
+#'      discover and interact with a specific widget instance.
+#' 
+#' @returns A \code{bpmnVisualization} Widget that will intelligently print itself into HTML in a variety of contexts
```
Comment (csouchet) [inline R/bpmnVisualization.R:30]:
```suggestion
#' @returns A \code{bpmn-visualization} Widget that will intelligently print itself into HTML in a variety of contexts 
```

File: R/bpmnVisualization.R [hunk 4]
```diff
@@ -49,21 +73,18 @@ display <- function(
...
 
-#' Shiny bindings for bpmnVisualization
-#'
-#' Output and render functions for using bpmnVisualization within Shiny
-#' applications and interactive Rmd documents.
+#' @title Shiny output binding for 'bpmnVisualization' HTML widget
 #'
+#' @name bpmnVisualization-shiny-output
+#' @description 
+#' Helper to create output function for using 'bpmnVisualization' HTML widget within Shiny applications and interactive Rmd documents.
```
Comment (csouchet) [inline R/bpmnVisualization.R:80]:
```suggestion
#' Helper to create output function for using \code{bpmn-visualization} HTML widget within Shiny applications and interactive Rmd documents.
```

File: R/bpmnVisualization.R [hunk 5]
```diff
@@ -49,21 +73,18 @@ display <- function(
...
 #'   \code{'400px'}, \code{'auto'}) or a number, which will be coerced to a
 #'   string and have \code{'px'} appended.
-#' @param expr An expression that generates a bpmnVisualization
-#' @param env The environment in which to evaluate \code{expr}.
-#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
-#'   is useful if you want to save an expression in a variable.
-#'
-#' @name bpmnVisualization-shiny
+#'   
+#' @returns An output function that enables the use of the 'bpmnVisualization' widget within Shiny applications.
```
Comment (csouchet) [inline R/bpmnVisualization.R:87]:
```suggestion
#' @returns An output function that enables the use of the \code{bpmn-visualization} widget within Shiny applications.
```

File: R/bpmnVisualization.R [hunk 6]
```diff
@@ -80,7 +101,19 @@ bpmnVisualizationOutput <- function(
   )
 }
 
-#' @rdname bpmnVisualization-shiny
+#' @title Shiny render binding for 'bpmnVisualization' HTML widget
```
Comment (csouchet) [inline R/bpmnVisualization.R:104]:
```suggestion
#' @title Shiny render binding for \code{bpmn-visualization} HTML widget
```

File: R/bpmnVisualization.R [hunk 7]
```diff
@@ -80,7 +101,19 @@ bpmnVisualizationOutput <- function(
...
+#' 
+#' @rdname bpmnVisualization-shiny-render
+#' @description 
+#' Helper to create render function for using 'bpmnVisualization' HTML widget within Shiny applications and interactive Rmd documents.
```
Comment (csouchet) [inline R/bpmnVisualization.R:108]:
```suggestion
#' Helper to create render function for using \code{bpmn-visualization} HTML widget within Shiny applications and interactive Rmd documents.
```

File: R/bpmnVisualization.R [hunk 8]
```diff
@@ -80,7 +101,19 @@ bpmnVisualizationOutput <- function(
...
+#' 
+#' @param expr An expression that generates a 'bpmnVisualization' HTML widget
```
Comment (csouchet) [inline R/bpmnVisualization.R:110]:
```suggestion
#' @param expr An expression that generates a \code{bpmn-visualization} HTML widget
```

File: R/bpmnVisualization.R [hunk 9]
```diff
@@ -80,7 +101,19 @@ bpmnVisualizationOutput <- function(
...
+#' @param env The environment in which to evaluate \code{expr}.
+#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
+#'   is useful if you want to save an expression in a variable.
+#'
+#' @returns A render function that enables the use of the 'bpmnVisualization' widget within Shiny applications.
```
Comment (csouchet) [inline R/bpmnVisualization.R:115]:
```suggestion
#' @returns A render function that enables the use of the \code{bpmn-visualization} widget within Shiny applications.
```

File: R/funs.R [hunk 10]
```diff
@@ -1,13 +1,17 @@
-#' create_overlay
+#' @title Create an overlay
 #'
+#' @name create_overlay
+#' @description 
 #' An overlay can be added to existing elements in the diagram.
+#' 
 #' See \code{overlays} argument in \code{\link{bpmnVisualization}} function.
```
Comment (csouchet) [inline R/funs.R:7]:
```suggestion
#' See \code{overlays} argument in \code{\link{display}} function.
```

**Comment by tbouffard** [comment]
⚠️ let's regenerate the Rd files and push them if needed. There is a risk that they are out-dated comparing to the source code

**Review by tbouffard — APPROVED** [review]
LGTM, good job (in pair 😄)!

**Pull request merged**